### PR TITLE
Move recaptcha keys configuration from per-page config data to shared…

### PR DESCRIPTION
**Pull request by cabrero**
… plugin config data.

When using recaptcha fileds in more than one form its better to share the api keys in the plugin config than repeating them in each form.md page file.

### CODE REVIEW

**Comment by flaviocopes** [comment]
Cannot merge it as it must be a backward compatible, or an update will break existing code. 
You can edit this proposal to check the page settings, and fallback to the global plugin configuration option if not found. 
It also needs an entry in the Blueprint to make it editable in the Admin Plugin

**Comment by cabrero** [comment]
You are totally right.
I've made the changes you suggested.
Also added some spanish translations.

File: form.php [hunk 1]
```diff
@@ -97,8 +97,11 @@ public function onFormProcessed(Event $event)
         switch ($action) {
             case 'captcha':
                 // Validate the captcha
+                $secret = $params['recatpcha_secret'];
+                if (empty($secret))
```
Comment (flaviocopes) [inline form.php]: Can you put a parentheses here? Just for clarity

File: blueprints.yaml [hunk 2]
```diff
@@ -54,4 +54,19 @@ form:
...
+    recaptcha:
+      type: section
+      title: PLUGIN_FORM.RECAPTCHA
+
+      fields:
+        recaptcha.site_key:
+          type: text
+          label: PLUGIN_FORM.RECAPTCHA_SITE_KEY
+          help: PLUGIN_FORM.RECAPTCHA_SITE_KEY_HELP
+          default: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI'
```
Comment (flaviocopes) [inline blueprints.yaml]: This should probably be an empty value (same for `secret_key`)

Comment (cabrero) [inline blueprints.yaml]: Those are the values for testing keys. But certainly that could be a little bit confusing.
Better go for empty default values.

**Comment by cabrero** [comment]
Improved coding style and moved to empty default api key values.

File: form.php [hunk 3]
```diff
@@ -97,8 +97,12 @@ public function onFormProcessed(Event $event)
         switch ($action) {
             case 'captcha':
                 // Validate the captcha
+                $secret = $params['recatpcha_secret'];
+                if (empty($secret)) {
+                    $secret = $this->config->get('plugins.form.recaptcha.secret');
```
Comment (flaviocopes) [inline form.php]: Could not get this to save, it must be `.secret_key` not `.secret`. After you update this, good to merge.

**Comment by rhukster** [comment]
I've manually merged this.

**Pull request closed without merging**